qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java

```diff
@@ -508,4 +514,49 @@ public void allowsNonTransientFields() throws Exception {
         ).validate();
     }
 
+    /**
+     * PMDValidator can prohibit public static methods.
+     * @throws Exception If something wrong happens inside.
+     */
+    @Test
+    public void prohibitsPublicStaticMethods() throws Exception {
+        new PMDAssert(
+            "StaticPublicMethod.java",
+            Matchers.is(false),
+            Matchers.containsString(PMDValidatorTest.STATIC_METHODS)
+        ).validate();
+    }
+
+    /**
+     * PMDValidator can allow public static void main(String...args) method.
+     * @throws Exception If something wrong happens inside.
+     */
+    @Test
+    public void allowsPublicStaticMainMethod() throws Exception {
+        new PMDAssert(
+            "StaticPublicVoidMainMethod.java",
+            Matchers.is(true),
+            Matchers.not(
+                Matchers.containsString(PMDValidatorTest.STATIC_METHODS)
+            )
+        ).validate();
+    }
+
+    /**
+     * PMDValidator can allow JUnit public static methods marked with:
+     * BeforeClass annotation.
+     * AfterClass annotation.
+     * Parameterized.Parameters annotation.
+     * @throws Exception If something wrong happens inside.
+     */
+    @Test
+    public void allowsJunitFrameworkPublicStaticMethods() throws Exception {
+        new PMDAssert(
+            "JunitStaticPublicMethods.java",
+            Matchers.is(true),
+            Matchers.not(
+                Matchers.containsString(PMDValidatorTest.STATIC_METHODS)
+            )
+        ).validate();
+    }
```

Comment:
@HDouss We have another PMD rule UseVarargs, that restricts us declare `public static void main(String args[])`. There is no need to check array as parameter.